### DISCUSSION TITLE
Base class format in classes

### DISCUSSION THREAD

**Discussion by quattro** [mkdocstrings/griffe#228] [Q&A] [answered]
Hi all,

I'm using mkdocstrings with the new python module which is based on griffe. I've found it produces wonderful documentation, but I'm curious what I would need to change (or add) in order to later the visual display of classes. For example, in pytkdocs, I like how base classes are displayed inline like

<img width="281" alt="image" src="https://github.com/mkdocstrings/griffe/assets/58283/f93ea2f8-0c79-4065-b7c6-64d2defdeb6f">,

rather than separately below the class section, like `Bases: Object`.

Would it require coding a custom extension, or is this something that can be done with a configuration.

Cheers,
Nick

**Comment by quattro** [comment]
Ah, is done through customization of the html template jinja block?

**Reply by pawamoy** [reply, marked as answer]
Hello, yes, you can achieve this with Jinja templates! See these two links:

- https://mkdocstrings.github.io/usage/theming/#templates
- https://mkdocstrings.github.io/python/usage/customization/#templates

You'd simply override the `class.html` template to change how bases are displayed. Something like:

```django
{% extends "_base/class.html" %}

{# Append bases at the end of the heading. #}
{% block heading %}
  {{ super() }}
  {%- if config.show_bases and class.bases- %}
    ({%- for expression in class.bases -%}
       <code>{% include "expression.html" with context %}</code>{% if not loop.last %}, {% endif %}
     {% endfor -%})
   {%- endif -%}
{% endblock %}

{# Empty the original "bases" block. #}
{% block bases %}{% endblock %}
```

Note that such customization will not be compatible with other options like `merge_init_into_class`.
